src/odemis/odemisd/modelgen.py

```diff
@@ -842,10 +878,11 @@ def get_instantiables(self, instantiated=None):
                 if psuname not in instantiated:
                     logging.debug("Component %s is not instantiable yet", n)
                     continue
-            for cname in attrs.get("children", {}).values():
-                childat = self.ast[cname]
+            references = attrs.get("children", {}).values() + attrs.get("dependencies", {}).values()
```

Comment:
Please add a comment explaining that in theory, we only care about the dependencies, but to handle legacy code which have their dependencies as "children", we also include children, and check if the creator is not the component.
Alternatively, you could write more explicitly:
```
dep = list(attrs.get("dependencies", {}).values())
deps += [c for c in attrs.get("children", {}).values() if self.ast[c].get("creator") != n]
for name in deps:
    if name not in instantiated:
        break
```
It'd make the code closer from the "final" code when the legacy support is removed.